nipype/utils/config.py

```diff
@@ -172,3 +173,52 @@ def update_matplotlib(self):
     def enable_provenance(self):
         self._config.set('execution', 'write_provenance', 'true')
         self._config.set('execution', 'hash_method', 'content')
+
+    def get_display(self):
```

Comment:
If I understand it, if this has not been called before workflow runtime, then each node will get [a copy of `config`](https://github.com/nipy/nipype/blob/ea62ca6ea9a788d6f98c3a5a57b582e2b383280f/nipype/pipeline/engine/nodes.py#L274-L277) and thus start up its own `Xvfb` object.

And I don't see anywhere where it's stopped.

Reply:
The first time you call it, one display will be locked. As a matter of fact, when I first wrote the tests (https://github.com/oesteban/nipype/blob/fix/VirtualDisplay/nipype/utils/tests/test_config.py#L9-L49) I realized I had to manally reset the config object all the times because once it is called it is fixed to that display. Therefore, all those `config._display = None` and those re-setting/removing the nipype config setting.

Does this reply to your question?

Reply:
Okay. I meant to put at the end of the first sentence: "Is this the desired behavior?" Sounds like it is.

Similarly, the fact that [`self._display.stop()`](https://github.com/nipy/nipype/pull/2203/files#diff-1cce190a2f31ba127aae1067c3860abfL1035) is no longer called anywhere is intentional?

Reply:
Yes, that was actually the problem: nipype would spawn an endless number of displays.

Well, it is probably nice to add a `self._display.stop()` at the end when running a workflow. I'm not sure though about how to do that when running bare interfaces.

Reply:
Ok, added a callback when python exists the session :D.